backend/drm/drm.c

```diff
@@ -643,14 +643,13 @@ static bool wlr_drm_connector_set_cursor(struct wlr_output *output,
 		wlr_texture_upload_pixels(plane->wlr_tex, WL_SHM_FORMAT_ARGB8888,
 			stride, width, height, buf);
 
-		glViewport(0, 0, plane->surf.width, plane->surf.height);
-		glClearColor(0.0, 0.0, 0.0, 0.0);
-		glClear(GL_COLOR_BUFFER_BIT);
+		struct wlr_renderer *rend = plane->surf.renderer->wlr_rend;
+		wlr_renderer_begin(rend, plane->surf.width, plane->surf.height);
+		wlr_renderer_clear(rend, (float[]){ 0.0, 0.0, 0.0, 0.0 });
+		wlr_render_texture(rend, plane->wlr_tex, plane->matrix, 0, 0, 1.0f);
+		wlr_renderer_end(rend);
 
-		wlr_render_texture(plane->surf.renderer->wlr_rend, plane->wlr_tex,
-			plane->matrix, 0, 0, 1.0f);
-
-		glFinish();
+		// TODO: remove these raw GL calls
```

Comment:
Didn't I originally write a function for this?

Reply:
Yeah, but it's reading pixel lines one by one in a loop. I wonder if I:
* Should use it anyway
* Should add a `bool invert_y` parameter to be able to use only one GL call
* Should keep it as is until there's a better solution

Reply:
There is a way it can be done in one call (GL_PACK_ROW_LENGTH), but as usual, it requires a version later than GLES2 (and is not available via extensions), so I was stuck doing it the stupid, inefficient way.

Reply:
Yeah. But you would still need to invert the y axis after that right? I didn't find a way to do this with GLES3 (it's possible to do so when rendering, not when reading pixels). Maybe we should make `wlr_renderer_read_pixels` always read pixels with an inverted y axis?

Anyway, I've pushed a commit to use the inefficient way.